iota/commands/core/get_balances.py

```diff
@@ -44,19 +45,50 @@ def __init__(self):
                     f.Min(0) |
                     f.Max(100) |
                     f.Optional(default=100),
+
+                'tips':
+                    f.Array | f.FilterRepeater(
+                        f.Required |
+                        Trytes(TransactionHash) |
+                        f.Unicode(encoding='ascii', normalize=False),
+                    )
             },
 
             allow_missing_keys={
-                'threshold',
+                'threshold', 'tips',
             },
         )
 
+    def _apply(self, value):
+        value = super(GetBalancesRequestFilter, self)._apply(
+            value
+        )  # type: dict
+
+        if self._has_errors:
+            return value
+
+        # Remove null search terms.
+        # Note: We will assume that empty lists are intentional.
+        search_terms = {
+            term: query
+            for term, query in iteritems(value)
+            if query is not None
+        }
+
+        return search_terms
+
 
 class GetBalancesResponseFilter(ResponseFilter):
     def __init__(self):
         super(GetBalancesResponseFilter, self).__init__({
             'balances': f.Array | f.FilterRepeater(f.Int),
 
-            'milestone':
-                f.ByteString(encoding='ascii') | Trytes(Address),
```

Comment:
Note that this is a potential backwards incompatible change. `milestone` was removed in IRI version 1.4.1.6 (https://github.com/iotaledger/iri/blob/v1.4.1.6/src/main/java/com/iota/iri/service/dto/GetBalancesResponse.java) which was released on the 6th of January 2018. Since this was more than one and a half years ago I think it is justified to remove it here too.

Reply:
👍🏻 Sounds good to me.